### PR TITLE
Enable CI with Travis

### PR DESCRIPTION
**Pull request by alexeyklyukin**
Add Travis CI configuration to run a. simple backup/restore test.

The test script has been changed to return 1 on failure, do not assume localhost connections (as Travis does not allow them without the password), instead connecting to the unix socket that is autodetected in the script, and increase the running time to 30s instead of 5 to catch less frequent errors. It should still run locally without any modifications

The code has been modified to honor PGHOST, PGPORT, PGDATABASE and PGUSER variables that are passed from the Travis Configuration.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -43,6 +45,8 @@ func New(filename string) (*Config, error) {
 	if err := yaml.NewDecoder(fp).Decode(&cfg); err != nil {
 		return nil, fmt.Errorf("could not decode config file: %v", err)
 	}
+	// work around pgx ignoring common Postgres variables
+	utils.InjectPGEnvironment(&cfg.DB)
```
[the remark this answers is not in the document]
Comment (alexeyklyukin) [reply pkg/config/config.go]: why, or why it's not something they do by default...

File: pkg/config/config.go [hunk 2]
```diff
@@ -36,13 +36,25 @@ func New(filename string) (*Config, error) {
 
 	fp, err := os.Open(filename)
 	if err != nil {
-		return nil, fmt.Errorf("could not open config file: %v", err)
+		return nil, fmt.Errorf("could not open cfg file: %v", err)
 	}
-	defer fp.Close()
+	defer func() {
+		if err := fp.Close(); err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "could not close file: %v", err)
```
Comment (ikitiki) [inline pkg/config/config.go]: keep it just `fmt.Fprintf(os.Stderr, "could not close file: %v", err)`

Comment (alexeyklyukin) [reply pkg/config/config.go]: that would make linters unhappy.

Comment (ikitiki) [reply pkg/config/config.go]: I think we can live with unhappy linter here

Comment (alexeyklyukin) [reply pkg/config/config.go]: or we should fix that as well; just not as part of this PR :-)

Comment (ikitiki) [reply pkg/config/config.go]: We'll fix that in the next PR, as for now I'd keep it code homogeneous and remove `_, _ = ` from this PR

**Pull request merged**